c_emulator/riscv_sim.c

```diff
@@ -337,6 +340,16 @@ char *process_args(int argc, char **argv)
       sailcov_file = strdup(optarg);
       break;
 #endif
+    case 'B':
+      block_size = atol(optarg);
+      if (((block_size & 7) == 0) && (block_size < 4096)) {
```

Comment:
I can't obviously find either of these restrictions in the spec; where have these come from? 4096 seems pretty unproblematic, that's a whole page, things just get awkward when you cover multiple pages, but the PMP already makes that an issue.